kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/RollingUpdateProcessServiceIntegrationTest.java

```diff
@@ -0,0 +1,296 @@
+
+            assertEquals(USER_MARY, variables.get("test"));
+            assertEquals(12345, variables.get("number"));
+            assertEquals(1, ((List) variables.get("list")).size());
+            assertEquals("item", ((List) variables.get("list")).get(0));
+            assertEquals(USER_JOHN, valueOf(variables.get("person"), "name"));
+            assertEquals(TestConfig.getUsername(), variables.get("initiator"));
+        } finally {
+            if (processInstanceId != null) {
+                processClient.abortProcessInstance(CONTAINER_ALIAS, processInstanceId);
+            }
+        }
+    }
+
+    @Test
+    public void testStartProcessInDifferentDeploymentWithAlias() throws Exception {
+
+        Object person = createPersonInstance(USER_JOHN);
+
+        Map<String, Object> parameters = new HashMap<String, Object>();
+        parameters.put("test", USER_MARY);
+        parameters.put("number", new Integer(12345));
+
+        List<Object> list = new ArrayList<Object>();
+        list.add("item");
+
+        parameters.put("list", list);
+        parameters.put("person", person);
+        Long processInstanceIdV1 = null;
+        Long processInstanceIdV2 = null;
+        try {
+            processInstanceIdV1 = processClient.startProcess(CONTAINER_ALIAS, PROCESS_ID_EVALUATION, parameters);
+
+            assertNotNull(processInstanceIdV1);
+            assertTrue(processInstanceIdV1.longValue() > 0);
+
+            ProcessInstance processInstance = processClient.getProcessInstance(CONTAINER_ALIAS, processInstanceIdV1);
+            assertNotNull(processInstance);
+            assertEquals(CONTAINER_ID, processInstance.getContainerId());
+
+            List<TaskSummary> tasks = taskClient.findTasksAssignedAsPotentialOwner(USER_YODA, 0, 10);
+            assertEquals(1, tasks.size());
+            TaskSummary task = tasks.get(0);
+            assertEquals(CONTAINER_ID, task.getContainerId());
+
+            taskClient.completeAutoProgress(CONTAINER_ALIAS, task.getId(), USER_YODA, null);
+
+            createExtraContainer();
+
+            processInstanceIdV2 = processClient.startProcess(CONTAINER_ALIAS, PROCESS_ID_EVALUATION, parameters);
+
+            assertNotNull(processInstanceIdV2);
+            assertTrue(processInstanceIdV2.longValue() > 0);
+
+            processInstance = processClient.getProcessInstance(CONTAINER_ALIAS, processInstanceIdV2);
+            assertNotNull(processInstance);
+            assertEquals(CONTAINER_ID_101, processInstance.getContainerId());
+
+            tasks = taskClient.findTasksAssignedAsPotentialOwner(USER_YODA, 0, 10);
+            assertEquals(1, tasks.size());
+            task = tasks.get(0);
+            assertEquals(CONTAINER_ID_101, task.getContainerId());
+
+            taskClient.completeAutoProgress(CONTAINER_ALIAS, task.getId(), USER_YODA, null);
+        } finally {
+            if (processInstanceIdV1 != null) {
```

Comment:
I think aborting processes could be done in a separate method with variable number of parameters, or even better - moved to a `@Before` method to decouple test scenario from clean up code and avoid repetition.